rust-lang/cargo
# refactor: git source cleanup

### What does this PR try to resolve?

I found some places can do minor refactors during documenting those items last week.

### How should we test and review this PR?

Commit by commit. Each has its own explanation.

I believe the first four is not controversial. However, the last three do need to scrutinize

* 45e3a8626780aeff2936b90de99144e39d59ef44 — I can't see why this need to implement `Serialize`. Did I miss some important contexts?
* 8ab4b686d2da7eee269fa69fab2b846a860794cd — the `repo` and `path` should always be in sync so there is no need to have two arguments. However, `git2::Repository::path()` force UTF8 path on Windows. Will it cause any problem? I guess it may not since we already use `repo.path()` somewhere in the codebase.
* e2c61e98136ca2b9314e1e78a9ca6f657e283c91 — This should have no behavioral change but I need a second pair of eyes to verify that.

### Additional information
